kernel/src/hil/crc.rs

```diff
@@ -0,0 +1,35 @@
+// Generic interface for CRC computation
+
+use returncode::ReturnCode;
+
+#[derive(Copy, Clone)]
+pub enum Polynomial {
+	CCIT8023,   // Polynomial 0x04C11DB7
```

Comment:
I don't understand enough about CRC, but there is a big matrix of options, and it seems like the SAM4L chooses a specific default: (reverse data, initial value 0xffffffff, don't reverse out data and don't xor out). Is there a way we can generalize the interface (other chips might choose other things), say by having the hil promise particular standard CRCs and implement whatever differences between that and the hardware under the hood?